pkg/service/dbaas/service.go

```diff
@@ -116,40 +127,48 @@ func (s *Service) fetchManagedDBaaS(ctx *Context) error {
 		}
 
 		for _, managedResource := range managedResources.Items {
-			dbaasDetail := Detail{
-				DBName: managedResource.GetName(),
-				Type:   groupVersionResource.Resource,
-			}
-			if organization, exist := managedResource.GetLabels()[service.OrganizationLabel]; exist {
-				dbaasDetail.Organization = organization
-			} else {
-				// cannot get organization from DBaaS
-				log.Info("Organization label is missing in DBaaS service, skipping...",
-					"label", service.OrganizationLabel,
-					"dbaas", managedResource.GetName())
-				continue
-			}
-			if namespace, exist := managedResource.GetLabels()[service.NamespaceLabel]; exist {
-				dbaasDetail.Namespace = namespace
-			} else {
-				// cannot get namespace from DBaaS
-				log.Info("Namespace label is missing in DBaaS, skipping...",
-					"label", service.NamespaceLabel,
-					"dbaas", managedResource.GetName())
+			dbaasDetail := findDBaaSDetailInNamespacesMap(managedResource, groupVersionResource, namespaces, log)
+			if dbaasDetail == nil {
 				continue
 			}
-			log.V(1).Info("Added namespace and organization to DBaaS",
-				"dbaas", managedResource.GetName(),
-				"namespace", dbaasDetail.Namespace,
-				"organization", dbaasDetail.Organization)
-			dbaasDetails = append(dbaasDetails, dbaasDetail)
+			dbaasDetails = append(dbaasDetails, *dbaasDetail)
 		}
 	}
 
 	ctx.dbaasDetails = dbaasDetails
 	return nil
 }
 
+func findDBaaSDetailInNamespacesMap(managedResource unstructured.Unstructured, groupVersionResource schema.GroupVersionResource, namespaces map[string]string, log logr.Logger) *Detail {
+	dbaasDetail := Detail{
+		DBName: managedResource.GetName(),
+		Type:   groupVersionResource.Resource,
+	}
+	if namespace, exist := managedResource.GetLabels()[service.NamespaceLabel]; exist {
```

Comment:
I noticed you are using a requirement object for the labels. I suppose with this approach you don't have to check if the lebel exists at this point, do you?

Reply:
I deleted the requirement for the labels, maybe with the rebase the changes can't be seeing 🤔

Reply:
yes, I'll remove the requirement object for the labels